autowarefoundation/autoware.universe
# feat: change planning factor behavior constants

## Description

Changed the velocity/steering factor type to string for extensibility of the module. Since it is an unreleased API, I don't think it will have a big impact, but the default_ad_api package converts it to the previous type to maintain compatibility.

## Related links

https://github.com/autowarefoundation/autoware_adapi_msgs/pull/45

## Tests performed

- Checkout https://github.com/autowarefoundation/autoware_adapi_msgs/pull/45
- Launch planning simulator
- Add VelocitySteeringFactorsPanel on rviz
- Set initialpose and goal
- Check if VelocitySteeringFactorsPanel works well

## Notes for reviewers

None

## Interface changes

https://github.com/autowarefoundation/autoware_adapi_msgs/pull/45.

## Effects on system behavior

None

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
